src/graphs/Dijkstra.rei

```diff
@@ -0,0 +1,19 @@
+type node = {
+    id: string,
+    neighbours: list(string),
+};
+
+type edge = {
+    tail: string,
+    head: string,
+    weight: int,
+};
+
+type directedGraph = list(node);
```

Comment:
why are you passing in the edges outside of the directedGraph?

Reply:
why don't we only specify the edges and construct the nodes inside the module?

Reply:
What if a node does not have any incident edges. I think we still need a list of nodes but without the `neighbours` field.